### PR TITLE
TST: Use more xfail instead of skip

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

In order to catch more behaviors that may be fixed in the future.

### CODE REVIEW

File: pandas/tests/resample/test_base.py [hunk 1]
```diff
@@ -98,11 +98,13 @@ def test_raises_on_non_datetimelike_index():
 
 @all_ts
 @pytest.mark.parametrize("freq", ["M", "D", "H"])
-def test_resample_empty_series(freq, empty_series_dti, resample_method):
+def test_resample_empty_series(freq, empty_series_dti, resample_method, request):
     # GH12771 & GH12868
 
-    if resample_method == "ohlc":
-        pytest.skip("need to test for ohlc from GH13083")
```
Comment (jbrockmendel) [inline pandas/tests/resample/test_base.py:105]: is the 13083 reference no longer relevant?

File: pandas/tests/io/test_parquet.py [hunk 2]
```diff
@@ -942,11 +942,17 @@ def test_timestamp_nanoseconds(self, pa):
...
 
-    def test_timezone_aware_index(self, pa, timezone_aware_date_list):
-        if not pa_version_under2p0:
-            # temporary skip this test until it is properly resolved
-            # https://github.com/pandas-dev/pandas/issues/37286
-            pytest.skip()
+    def test_timezone_aware_index(self, request, pa, timezone_aware_date_list):
+        if (
+            not pa_version_under2p0
+            and not timezone_aware_date_list.tzinfo == datetime.timezone.utc
```
Comment (jbrockmendel) [inline pandas/tests/io/test_parquet.py:948]: nitpick "!="

File: pandas/tests/io/json/test_ujson.py [hunk 3]
```diff
@@ -997,9 +997,11 @@ def test_dataframe_nested(self, orient):
         }
         assert ujson.decode(ujson.encode(nested, **kwargs)) == exp
 
-    def test_dataframe_numpy_labelled(self, orient):
+    def test_dataframe_numpy_labelled(self, orient, request):
         if orient in ("split", "values"):
-            pytest.skip("Incompatible with labelled=True")
```
Comment (jbrockmendel) [inline pandas/tests/io/json/test_ujson.py:1002]: @willayd is xfail the right mark for this?  i.e. is it something that needs attention

File: pandas/tests/groupby/test_groupby.py [hunk 4]
```diff
@@ -713,11 +713,8 @@ def test_ops_not_as_index(reduction_func):
     # GH 10355, 21090
     # Using as_index=False should not modify grouped column
 
-    if reduction_func in ("corrwith",):
-        pytest.skip("Test not applicable")
-
-    if reduction_func in ("nth", "ngroup"):
-        pytest.skip("Skip until behavior is determined (GH #5755)")
```
Comment (jbrockmendel) [inline pandas/tests/groupby/test_groupby.py:720]: is the GH#5755 reference no longer relevant?

File: pandas/tests/resample/test_base.py (hunk 1 above)
Comment (mroeschke) [reply pandas/tests/resample/test_base.py:105]: It's a closed issue, but I'll add back the reference.

File: pandas/tests/groupby/test_groupby.py (hunk 4 above)
Comment (mroeschke) [reply pandas/tests/groupby/test_groupby.py:720]: It's a closed issue, but I'll add back the reference.

**Pull request merged**